server/src/main/java/org/opensearch/search/aggregations/support/MultiValuesSource.java

```diff
@@ -59,12 +59,12 @@ public NumericMultiValuesSource(Map<String, ValuesSourceConfig> valuesSourceConf
             values = new HashMap<>(valuesSourceConfigs.size());
             for (Map.Entry<String, ValuesSourceConfig> entry : valuesSourceConfigs.entrySet()) {
                 final ValuesSource valuesSource = entry.getValue().getValuesSource();
-                if (valuesSource instanceof ValuesSource.Numeric == false) {
+                if (!(valuesSource instanceof ValuesSource.Numeric numericValuesSource)) {
                     throw new AggregationExecutionException(
                         "ValuesSource type " + valuesSource.toString() + "is not supported for multi-valued aggregation"
                     );
                 }
-                values.put(entry.getKey(), (ValuesSource.Numeric) valuesSource);
+                values.put(entry.getKey(), numericValuesSource);
```

Comment:
Looks interesting to me that `numericValuesSource` is accessible outside the scope of where it is defined!